src/iso19111/io.cpp

```diff
@@ -2963,8 +2963,24 @@ WKTParser::Private::buildConversion(const WKTNodeNNPtr &node,
     } else if (methodProps.getStringValue(IdentifiedObject::NAME_KEY, methodName) &&
         starts_with(methodName,  "PROJ-based operation method:")) {
         auto projString = methodName.substr(strlen("PROJ-based operation method: "));
-        if (starts_with(projString, "+proj=")) {
-            projString += geodeticCRS->datum()->ellipsoid()->exportToPROJString(PROJStringFormatter::create().get());
+        bool ellips_parameters_exist = (
+            projString.find("+a=") != std::string::npos
+            || projString.find("+ellps=") != std::string::npos
+        );
+        // TODO: skip this if ellipsoid parameters do not exist
+        // and ellipsoid is not passed in (need to find non nullable method)
+        // and the projection needs them to be properly defined
+        if (starts_with(projString, "+proj=")
```

Comment:
I'm unconfortable with this ad-hoc way of determinining if we can try the createFromPROJString() way. I believe this should be a post check, based on analyzing the initial PROJ string and the one got from projObjCrs->derivingConversion()->exportToPROJString().  PROJStringSyntaxParser could be used to do the tokenizing, but then some new code should be added to compare two tokenized PROJ string (to be independent for example of the order of parameters in a step)

Reply:
Okay, I will remove the checks and plan on adding in post-checks later.